modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js

```diff
@@ -270,16 +270,8 @@ const Modal = ({
 
 			eventHandlers.splice(0, eventHandlers.length);
 		};
-	}, [
-		customEvents,
-		eventHandlersRef,
-		id,
-		onClose,
-		onOpen,
-		onSelect,
-		processClose,
-		selectEventName,
-	]);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, []);
```

Comment:
Would be interesting to know which item(s) in the deps array was/were triggering unwanted updates. One that stands out is this one, `customEvents`, which has a default value, so if not passed, you'll get a different array every time the function is called. A safer default for that would be `null`, and you'd just need an `if (customEvents)` guard before using it. This in general is true with React components, that non-primitive defaults (common examples being functions, objects, arrays) can be a source of trouble for exactly this reason.

If you wanted to find out which elements in the array are the cause, you could do a comparison with `Object.is()`, which is what React is using internally to perform these checks.

In my experience I believe refs tend to be pretty stable, but I don't think there are any guarantees documented anywhere. I'd be interested to hear from anybody who has "caught" `useRef` returning an unstable value; at least playing around in CodeSandbox now I always get back the same thing in a trivial example. I remember looking in the React source some months back to figure that out, but I've forgotten so would need to look again...

The thing we actually care about is `eventHandlersRef.current` anyway; as you know, if you stick that in the array, the lint rule will tell you:

> React Hook React.useEffect has an unnecessary dependency: 'ref.current'. Either exclude it or remove the dependency array. Mutable values like 'ref.current' aren't valid dependencies because mutating them doesn't re-render the component.

FWIW, if there is any evidence that the ref is changing, I think I would just omit it from the list and if the linter complains, suppress it.

But looking at the [React source](https://github.com/facebook/react/blob/30b47103d4354d9187dc0f1fb804855a5208ca9f/packages/react-reconciler/src/ReactFiberHooks.new.js#L1197-L1210), I'd say this shows the ref to be memoized and stable, so it is very unlikely to be the problem, and it seems safe for it to stay in the array.

Reply:
I think it was `visible`

Reply:
> Would be interesting to know which item(s) in the deps array was/were triggering unwanted updates. 

The ones causing the unwanted updates are `customEvents` and `processClose`. You are right, the ref is safe.

Removing default for `customEvents` and wrapping usage in `if` did work! That's such a weird quirk, I would never have guessed it works that way :open_mouth:  

That leaves `processClose`. It's a hard nut to crack, we are changing state in it. Is there some magic to workaround that one?

If not, does it make sense to keep partial list of arguments? We would still have to keep the `// eslint-disable-next-line react-hooks/exhaustive-deps` line.

Reply:
> That leaves `processClose`. It's a hard nut to crack, we are changing state in it. Is there some magic to workaround that one?

As it turns out, reverting `useCallback` does the job.

With this, we don't need the rule exception. Let me push the changes.

Reply:
Pushed!

Reply:
> I think it was `visible`, based on my analysis at [#50 (comment)](https://github.com/liferay-frontend/liferay-portal/pull/50#issuecomment-649313914)

`visible` as an argument indeed would cause updates. That's why I removed it, by [reverting the commit](https://github.com/liferay-frontend/liferay-portal/pull/59/commits/b397c789f774ed42e390a5a2cdb873c6e6272613) where you added it.

Reply:
> visible as an argument indeed would cause updates. That's why I removed it, by reverting the commit where you added it.

😂 

Didn't realize I added it!!! This lint rule is waaaay toooo smart!! 😱

Reply:
> This lint rule is waaaay toooo smart!!

![smart](https://user-images.githubusercontent.com/7074/85759990-b3f92c80-b711-11ea-9c5a-09fdbe856309.gif)

> The ones causing the unwanted updates are `customEvents` and `processClose`. You are right, the ref is safe.

But `processClose` only because you got rid of the `useCallback`, right? (I haven't looked at your new changes yet, so I don't know if you added back the `useCallback`.)

... and reading further down the thread I see you discovered that.

> That leaves `processClose`. It's a hard nut to crack, we are changing state in it. Is there some magic to workaround that one?

I feel like we discussed `useCallback` on that one in a prior iteration, but I can't find the reference to it. 🤷

Reply:
> I feel like we discussed `useCallback` on that one in a prior iteration, but I can't find the reference to it. 🤷

This was it: https://github.com/wincent/liferay-portal/pull/295#discussion_r430402239